### ISSUE TITLE
blockScraper implementation

### ISSUE THREAD

**Issue by lurenss** [ScrapeGraphAI/Scrapegraph-ai#88]
**Is your feature request related to a problem? Please describe.**
A scraper pipeline capable of retrieve all the similar blocks in a page, like ecommerce, weather, fly websites

**Describe the solution you'd like**
I have found this paper https://www.researchgate.net/publication/261360247_A_Web_Page_Segmentation_Approach_Using_Visual_Semantics
It deals specifically wti this issue

**Describe alternatives you've considered**
nope

**Additional context**
![Screenshot 2024-04-27 at 15 04 05](https://github.com/VinciGit00/Scrapegraph-ai/assets/38807022/1f935cad-3201-4289-8ceb-d49b2874ebff)

**Comment by epage480** [comment]
Neat idea but would it be simpler to just group web elements with the same css tags? A computer vision approach seems a bit over-engineered.

**Comment by lurenss** [comment]
@epage480 This isn't a CV approach, it's a grouping similar object from HTML, if you want to help to implement this paper with us let me know, here the reference [A Web Page Segmentation Approach Using Visual Semantics](https://sci-hub.st/10.1587/transinf.e97.d.223)

**Comment by DiTo97** [comment]
@lurenss I would much rather focus on this paper, https://doi.org/10.1007/978-3-030-72240-1_5

much more recent and detailed comparison of all major web page segmentation algorithms.

tl;dr - microsoft's [VIPS](https://www.microsoft.com/en-us/research/publication/vips-a-vision-based-page-segmentation-algorithm/) algorithm is still the best one out there; you can find implementations in [java](https://github.com/tpopela/vips_java), [JS](https://github.com/webis-de/ecir21-an-empirical-comparison-of-web-page-segmentation-algorithms/blob/master/algorithms/vips/scripts/VIPSScript-1.0.0/vips.js) or [python](https://github.com/wushuartgaro/VipsPython).
